File: programmers/js/lev0/수 조작하기 2.js
function solution(numLog) {
  let answer = "";

  let tmp = 0;
  for (let i = 0; i < numLog.length - 1; i++) {
    tmp = numLog[i + 1] - numLog[i];
    if (tmp == 1) {
      answer += "w";
    } else if (tmp == -1) {
      answer += "s";
    } else if (tmp == 10) {
      answer += "d";
    } else {
      answer += "a";
    }
  }
  return answer;
}

File: programmers/java/lev0/수 조작하기 2.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;
class Solution {
    public String solution(int[] numLog) {
        Map<Integer, String> dict = new HashMap<>();
        
        dict.put(1,"w");
        dict.put(-1,"s");
        dict.put(10,"d");
        dict.put(-10,"a");
        
        String answer="";
        for(int i=1;i<numLog.length;i++){
            answer+=dict.get(numLog[i]-numLog[i-1]);
        }
        return answer;
    }
}